src/sendqrg.c

```diff
@@ -250,46 +243,46 @@ static int parse_rigconf() {
 }
 
 
-static void debug_tlf_rig() {
-    freq_t rigfreq;
-    int retcode;
-
-    sleep(10);
-
-    pthread_mutex_lock(&tlf_rig_mutex);
-    retcode = rig_get_freq(my_rig, RIG_VFO_CURR, &rigfreq);
-    pthread_mutex_unlock(&tlf_rig_mutex);
-
-    if (retcode != RIG_OK) {
-	show_rigerror("Problem with rig get freq", retcode);
-    } else {
-	shownr("freq =", (int) rigfreq);
-    }
-    sleep(10);
-
-    const freq_t testfreq = 14000000;	// test set frequency
+/* callback receiving hamlibs debug output */
+int rig_debug_cb(enum rig_debug_level_e lvl,
+		 rig_ptr_t user_data,
+		 const char *fmt,
+		 va_list ap) {
 
-    pthread_mutex_lock(&tlf_rig_mutex);
-    retcode = rig_set_freq(my_rig, RIG_VFO_CURR, testfreq);
-    pthread_mutex_unlock(&tlf_rig_mutex);
+    enum debuglevel level;
 
-    if (retcode != RIG_OK) {
-	show_rigerror("Problem with rig set freq", retcode);
-    } else {
-	showmsg("Rig set freq ok!");
+    /* convert hamlib debug levels into Tlfs ones */
+    switch (lvl) {
+	case RIG_DEBUG_ERR:
+	    level = TLF_DBG_ERR;
+	    break;
+	case RIG_DEBUG_WARN:
+	    level = TLF_DBG_WARN;
+	    break;
+	case RIG_DEBUG_VERBOSE:
+	    level = TLF_DBG_INFO;
+	    break;
+	case RIG_DEBUG_TRACE:
+	    level = TLF_DBG_DEBUG;
+	    break;
+	default:
+	    level = TLF_DBG_NONE;
     }
 
-    pthread_mutex_lock(&tlf_rig_mutex);
-    retcode = rig_get_freq(my_rig, RIG_VFO_CURR, &rigfreq);	// read qrg
-    pthread_mutex_unlock(&tlf_rig_mutex);
+    char *format = g_strdup_printf("Rig: %s", fmt);
+    char *msg = g_strdup_vprintf(format, ap);
+    debug_log(level, msg);
+    g_free(msg);
+    g_free(format);
+    return RIG_OK;
+}
 
-    if (retcode != RIG_OK) {
-	show_rigerror("Problem with rig get freq", retcode);
-    } else {
-	shownr("freq =", (int) rigfreq);
-	if (rigfreq != testfreq) {
-	    showmsg("Failed to set rig freq!");
-	}
+/* set hamlibs debug level and install callback if debug is active */
+void rig_debug_init() {
+    if (debug_is_active()) {
+	/* set hamlib debug level and install callback */
+	rig_set_debug(RIG_DEBUG_TRACE);
```

Comment:
Could we set the matching hamlib log level? This way we would avoid a lot of stuff in the callback: building `format` and `msg`, then first in `debug_log` figure out that all is unneeded and free them.

Reply:
Could be done. Did checks before to guess the load. Additional load was below 1 % CPU time.
But I will have a look into it.

Reply:
How about the last commit?